lib-multisrc/liliana/src/eu/kanade/tachiyomi/multisrc/liliana/Liliana.kt

```diff
@@ -0,0 +1,367 @@
+        val filters = mutableListOf<Filter<*>>()
+
+        if (genreData.isNotEmpty()) {
+            filters.add(GenreFilter(genreName, genreData))
+        }
+        if (chapterCountData.isNotEmpty()) {
+            filters.add(ChapterCountFilter(chapterCountName, chapterCountData))
+        }
+        if (statusData.isNotEmpty()) {
+            filters.add(StatusFilter(statusName, statusData))
+        }
+        if (genderData.isNotEmpty()) {
+            filters.add(GenderFilter(genderName, genderData))
+        }
+        if (sortData.isNotEmpty()) {
+            filters.add(SortFilter(sortName, sortData))
+        }
+        if (filters.size < 5) {
+            filters.add(0, Filter.Header("Press 'reset' to load more filters"))
+        } else {
+            filters.add(0, Filter.Header("NOTE: Ignored if using text search!"))
+            filters.add(1, Filter.Separator())
+        }
+
+        return FilterList(filters)
+    }
+
+    private val scope = CoroutineScope(Dispatchers.IO)
+
+    protected fun launchIO(block: suspend () -> Unit) = scope.launch { block() }
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        description = document.selectFirst("div#syn-target")?.text()
+        thumbnail_url = document.selectFirst(".a1 > figure img")?.imgAttr()
+        title = document.selectFirst(".a2 header h1")?.text()?.trim() ?: "N/A"
+        genre = document.select(".a2 div > a[rel='tag'].label").joinToString { it.text() }
+        author = document.selectFirst("div.y6x11p i.fas.fa-user + span.dt")?.text()?.takeUnless {
+            it.equals("updating", true)
+        }
+        status = document.selectFirst("div.y6x11p i.fas.fa-rss + span.dt").parseStatus()
+    }
+
+    private fun Element?.parseStatus(): Int = when (this?.text()?.lowercase()) {
+        "ongoing", "đang tiến hành", "進行中" -> SManga.ONGOING
+        "completed", "hoàn thành", "完了" -> SManga.COMPLETED
+        "on-hold", "tạm ngưng", "保留" -> SManga.ON_HIATUS
+        "canceled", "đã huỷ", "キャンセル" -> SManga.CANCELLED
+        else -> SManga.UNKNOWN
+    }
+
+    // ============================== Chapters ==============================
+
+    override fun chapterListSelector() = "ul > li.chapter"
+
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        element.selectFirst("time[datetime]")?.also {
+            date_upload = it.attr("datetime").toLongOrNull()?.let { it * 1000L } ?: 0L
+        }
+        with(element.selectFirst("a")!!) {
+            name = text()
+            setUrlWithoutDomain(attr("abs:href"))
+        }
+    }
+
+    // =============================== Pages ================================
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val document = client.newCall(GET(baseUrl + chapter.url, headers)).execute().asJsoup()
```

Comment:
request methods should ideally only return requests.

also it appears chapter ids can be saved from the main manga page
![image](https://github.com/keiyoushi/extensions-source/assets/48650614/8f9746a3-5ed4-4e7a-81ad-359464f979bf)